### PR TITLE
Add `regexp/confusing-quantifier` rule

### PR DESCRIPTION
**Pull request by ota-meshi**

### CODE REVIEW

**Comment by RunDevelopment** [comment]
@ota-meshi It seems like you were faster than me. I was busy moving the AST analysis code (`ast-util.ts`) into its own library (and making some improvements along the way). I also use these functions in some other projects too, so this step was long overdue.

I'll start moving over this weekend, so you won't have to do all the work yourself :)

**Comment by ota-meshi** [comment]
I'm looking forward to the release of that package😀

**Comment by RunDevelopment** [comment]
I'll try to publish it today. I'll help move over the clean-regex rules this weekend.

**Comment by ota-meshi** [comment]
If I have time, I'm checking and working on clean-regex rules based on #91. Feel free to comment if you have any questions😁

File: lib/rules/confusing-quantifier.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    getQuantifierOffsets,
+    getRegexpLocation,
+    quantToString,
+} from "../utils"
+import { isPotentiallyEmpty } from "regexp-ast-analysis"
+
+export default createRule("confusing-quantifier", {
+    meta: {
+        docs: {
+            description: "disallow about confusing quantifiers",
```
Comment (RunDevelopment) [inline lib/rules/confusing-quantifier.ts:15]: Typo

```suggestion
            description: "disallow confusing quantifiers",
```

**Comment by ota-meshi** [comment]
@RunDevelopment  Thank you for your review!
I changed the rule description.

**Review by RunDevelopment — APPROVED** [review]

**Pull request merged**